### PR TITLE
backport: bitcoin#13072, #14096, #14161, #15831, #16250, #16251, #17049, #17056, #17138, #18032 (rpc_createmultisig.py and some extra)

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
These changes are backports from bitcoin v20 and earlier.

## What was done?
This PR includes backports from bitcoin somehow related to new functional test rpc_createmultisig.py include some prior changes. 
 - bitcoin#17049 
 - bitcoin#14096 
 - bitcoin#14161 
 - bitcoin#13072 
 - bitcoin#15831 
 - bitcoin#16250 
 - bitcoin#16251 
 - bitcoin#17056 
 - bitcoin#18032 
 - bitcoin#17138 

Beside them were added missing changes from backports that already marked as "done": 
 - bitcoin#16026
 - bitcoin#17675
 - bitcoin#13541
 - bitcoin#14180
These changes were missing because the functional test `rpc_createmultisig.py` is introduced only now within backport bitcoin#13072

## How Has This Been Tested?
Run unit and functional tests.

## Breaking Changes
No breaking changes

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -3149,6 +3148,7 @@ static bool IsCurrentForAntiFeeSniping(interfaces::Chain& chain)
  */
 static uint32_t GetLocktimeForNewTransaction(interfaces::Chain& chain)
 {
+    uint32_t const height = chain.getHeight().value_or(-1);
```
Comment (PastaPastaPasta) [inline src/wallet/wallet.cpp:3151]: 17138 how is this related?

Comment (knst) [reply src/wallet/wallet.cpp:3151]: These 2 are aliens and just stayed in this branch after all rebases:
https://github.com/bitcoin/bitcoin/pull/17138
https://github.com/bitcoin/bitcoin/pull/17049
Not related to `rpc_createmultisig.py`

Comment (PastaPastaPasta) [reply src/wallet/wallet.cpp:3151]: But I don't see the changes you attribute to 17138 in that commit

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit
